server/Arcadia.Assistant.Organization.Abstractions/VacationApprovalsChecker.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace Arcadia.Assistant.Organization.Abstractions
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    using Akka.Actor;
+    using Akka.DI.Core;
+
+    using OrganizationRequests;
+
+    public abstract class VacationApprovalsChecker : UntypedActor, ILogReceive
+    {
+        public static Props GetProps()
+        {
+            return Context.DI().Props<VacationApprovalsChecker>();
+        }
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case GetNextVacationRequestApprover msg:
+                    this.GetNextApprover(msg.EmployeeId, msg.ExistingApprovals)
+                        .PipeTo(this.Sender, success: r => new GetNextVacationRequestApprover.Response(r));
```

Comment:
add a Faliure? So it won't fail with a timeout when database fails with an error

Reply:
I think now it sends a generic Failure message back to sender, which is not handled, and your code waits for Response than never happens and things go south :)

Reply:
Like you tried with Response, Success: Response, Failure: Response before.

Reply:
Then you can handle it. I think now you just await it via Ask? So it'll fail with timeout? Here, you'll be able to handle it properly

Reply:
Yes, I understand. But what's next? How application will know about next approver, if this request returned an error?